.github/workflows/image-build.yaml

```
@@ -41,3 +57,18 @@ jobs:
     secrets:
       registry_username: ${{ secrets.QUAY_PUBLISH_ROBOT }}
       registry_password: ${{ secrets.QUAY_PUBLISH_TOKEN }}
+  
+  # generic provider requires go dep provider we built in matrix and needs to be sequential
+  generic-external-provider-build:
```

I think you want this to wait for [image-build to be successful](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#defining-prerequisite-jobs).

```suggestion
  generic-external-provider-build:
    needs: image-build
```